### PR TITLE
Rename Page to Menu Item

### PR DESCRIPTION
**Pull request by tonypartridge**
Pull Request for Issue #17408 .

Page is not a word natively used in Joomla! we should stick to standard convention of Menu Item.

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 37cd2f329ea03ea994876e874194a92e12e3c57c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17409">issues.joomla.org/tracker/joomla-cms/17409</a>.</sub>

File: administrator/language/en-GB/en-GB.com_modules.ini [hunk 1]
```diff
@@ -134,7 +134,7 @@ COM_MODULES_OPTION_POSITION_TEMPLATE_DEFINED="Template"
 COM_MODULES_OPTION_POSITION_USER_DEFINED="User"
 COM_MODULES_OPTION_SELECT_CLIENT="- Select Type -"
 COM_MODULES_OPTION_SELECT_MODULE="- Select Type -"
-COM_MODULES_OPTION_SELECT_PAGE="- Select Page -"
+COM_MODULES_OPTION_SELECT_MENU_ITEM="- Select Menu Item -"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_modules.ini]: Sort alpha order.

File: administrator/language/en-GB/en-GB.com_templates.ini [hunk 2]
```diff
@@ -184,7 +184,7 @@ COM_TEMPLATES_NEW_FILE_NAME="New File Name"
 COM_TEMPLATES_NEW_FILE_SELECT="Select a file type"
 COM_TEMPLATES_NEW_FILE_TYPE="File Type"
 COM_TEMPLATES_NO_TEMPLATE_SELECTED="No template selected."
-COM_TEMPLATES_OPTION_SELECT_PAGE="- Select Page -"
+COM_TEMPLATES_OPTION_SELECT_MENU_ITEM="- Select Menu Item -"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_templates.ini]: Sort alpha order.

File: administrator/language/en-GB/en-GB.com_modules.ini (hunk 1 above)
Comment (tonypartridge) [reply administrator/language/en-GB/en-GB.com_modules.ini]: Are you a real person or just a super computer?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7c342b9be11171af50346313e7880fb8f036bc93<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17409">issues.joomla.org/tracker/joomla-cms/17409</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 7c342b9be11171af50346313e7880fb8f036bc93<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17409">issues.joomla.org/tracker/joomla-cms/17409</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by mbabker** [comment]
Not RTC.  We have that crazy policy about not renaming language keys because they might be used elsewhere and blah blah blah, so the old strings have to be deprecated and kept until 4.0.

**Comment by ghost** [comment]
set back on "Pending"as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17409">issues.joomla.org/tracker/joomla-cms/17409</a>.</sub>

**Comment by brianteeman** [comment]
Agree with @mbabker although we don't usually bother to mark language strings as deprecated

**Comment by mbabker** [comment]
Well, unless we want to keep translating strings that stopped being used in core 5 users ago because nobody ever bothers to clean that stuff out, we gotta start doing things "right" at some point.

**Comment by tonypartridge** [comment]
How do you even do language string depreciations?

I can add them back in no problem! Just didn't think anyone would be using them.

**Comment by tonypartridge** [comment]
@mbabker what is the correct method to mark deprecated? Add to end of file with a deprecated 4.0 comment?

**Comment by Bakual** [comment]
In the past, we added a comment before the deprecated string with that deprecation message. That comment will show up in com_localise (but not in Crowdin).

Don't put the lines at the end of the file, keep them in the place they are.

**Comment by brianteeman** [comment]
For reference I have already removed a lot of unused strings in J4

On 4 Aug 2017 8:42 pm, "Thomas Hunziker" <notifications@github.com> wrote:

In the past, we added a comment before the deprecated string with that
deprecation message. That comment will show up in com_localise (but not in
Crowdin).

Don't put the lines at the end of the file, keep them in the place they are.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/joomla/joomla-cms/pull/17409#issuecomment-320324281>,
or mute the thread

**Comment by infograf768** [comment]
Just keeping the original string (with its constant) in the ini file makes no sense.
This will not be B/C because the original constant is not used anymore in the xml.
Why not just keep the original constant and just change its value?

**Comment by brianteeman** [comment]
> Just keeping the original string (with its constant) in the ini file makes no sense.

That is exactly what has been done in the past at your insistence (there are many examples of this)

> This will not be B/C because the original constant is not used anymore in the xml.

There is no B/C break as it is a new string that is being used. We are only keeping the old string available because of your insistence that it could be being used by a third party. (there are many examples of this)

**Comment by tonypartridge** [comment]
To me to keep the old string is silly if it's not used in the core and it's not a JGLOBAL string. It doesn't break anything and just looks a little ugly should someone be using a core component string which means they know they are doing it and can release a fix using the new string.

**Comment by brianteeman** [comment]
@tonypartridge agreed but thats what has been demanded in the past. i spent ages removing those unused strings for j4

**Comment by tonypartridge** [comment]
@mbabker I call for change! Any non JGLOBAL strings can be removed if not used in the core.

**Comment by infograf768** [comment]
> There is no B/C break as it is a new string that is being used. We are only keeping the old string available because of your insistence that it could be being used by a third party. (there are many examples of this)

My mistake, sorry indeed. Not awaken this morning...

It is true that lang packages can be updated (as they are proposed for any 3.x J version) while the J version is not and therefore the reason why we should keep old strings and deprecate them, JGLOBAL or NOT.

In this specific case though, the change in the value is small and keeping the same constant would be enough while changing the value imho.

**Comment by tonypartridge** [comment]
Whilst I agree with the change of the value and not the string people wouldn't be happy with the inconsistency for translations etc.

**Comment by infograf768** [comment]
There will always be inconsistencies in translations, (as well as in the en-GB reference language files).
It is good to correct as much as we can and do our best while we modify the en-GB in core.

1.TTs are volunteers and may or may not update their packs in time for a release + some disappear after a few major version packs. We should nevertheless keep as many languages as we can for the final users.
2. Also, as Brian wrote, 3pd may be using some existing strings in their extensions (in fact we should encourage them to do that more...), and not only lib or global ones

4.0 will be the occasion for a good cleaning and start from scratch for many languages, as we did after 1.5 and 2.5.

**Comment by tonypartridge** [comment]
Oh god at this rate I'll close the PR 😂.

Can we just get an official verdict on what's right

**Comment by brianteeman** [comment]
What you have now is correct

**Pull request merged**